app/models/manageiq/providers/ibm_power_hmc/infra_manager/host.rb

```diff
@@ -1,2 +1,123 @@
 class ManageIQ::Providers::IbmPowerHmc::InfraManager::Host < ::Host
+  def capture_metrics(counters, start_time = nil, end_time = nil)
+    samples = collect_samples(start_time, end_time)
+    process_samples(counters, samples)
+  end
+
+  def collect_samples(start_time, end_time)
+    ext_management_system.with_provider_connection do |connection|
+      connection.managed_system_metrics(
+        :sys_uuid => ems_ref,
+        :start_ts => start_time,
+        :end_ts   => end_time
+      )
+    rescue IbmPowerHmc::Connection::HttpError => e
+      $ibm_power_hmc_log.error("error getting performance samples for host #{ems_ref}: #{e}")
+      unless e.message.eql?("403 Forbidden") # TO DO - Capture should be disabled at Host level if PCM is not enabled
+        raise
+      end
+
+      []
+    end
+  end
+
+  def process_samples(counters, samples)
+    metrics = {}
+    samples.dig(0, "systemUtil", "utilSamples")&.each do |s|
```

Comment:
```suggestion
    samples.dig(0, "systemUtil", "utilSamples")&.each_with_object({}) do |s, metrics|
```
Then you don't need to have the extra `metrics` return down on `+34` it'll be the return from the block

Reply:
ok